src/cockpit/channels/filesystem.py

```diff
@@ -168,10 +228,21 @@ def delete(self, path: str, tag: 'str | None') -> str:
             os.unlink(path)
         return '-'
 
-    async def set_contents(self, path: str, tag: 'str | None', data: 'bytes | None', size: 'int | None') -> str:
+    async def set_contents(self, path: str, tag: 'str | None', data: 'bytes | None', size: 'int | None',
+                           attrs: 'FSReplaceAttrs | None') -> str:
         dirname, basename = os.path.split(path)
         tmpname: str | None
         fd, tmpname = tempfile.mkstemp(dir=dirname, prefix=f'.{basename}-')
+
+        def chown_if_required(fd: 'int'):
+            if attrs is None:
+                return
+
+            if attrs.uid is not None and attrs.gid is not None:
+                os.fchown(fd, attrs.uid, attrs.gid)
+            elif attrs.uid is not None:
+                os.fchown(fd, attrs.uid, attrs.uid)
```

Comment:
This is wrong. You translate an user/group name into an UID, but it's by no means guaranteed that their corresponding IDs are equal.
```
❱❱❱ getent passwd saslauth
saslauth:x:998:76:Saslauthd user:/run/saslauthd:/sbin/nologin

❱❱❱ getent group saslauth
saslauth:x:76:saslauth
```

Specify unchanged attributes as  "-1". https://linux.die.net/man/2/fchown

Reply:
My idea here was to support `chown jelle:` which sets both uid/gid to `jelle`. If not desired we can leave it out and throw a ProtocolError when they dont match

Reply:
Ah yes, that's fine -- but then you need something like `getpwuid(attrs.uid).gid)`. Please cover this in tests.

Reply:
I've thought about this again, and prefer passing `-1` when either owner or group is not provided. That should be less magic then only passing uid also changing the group.